File: src/python/main/other/rich_json_class_mapping.py
from main.helper.rich_json_logger import RichJsonLogger

_RICH_JSON_CLASS_MAPPING = {}
logger = RichJsonLogger.logger


def add_class_mappings(class_mappings):
    """
    Adds the given class mappings.

    :param class_mappings: A dictionary of name-to-class mappings.
    """
    if not isinstance(class_mappings, dict):
        return

    for name, class_type in class_mappings.items():
        add_class_mapping(name, class_type)


def add_class_mapping(name, class_type):
    """
    Adds the given class to the mapping table.

    :param name: The string name representing the class.
    :param class_type: The class reference itself.
    :raises ValueError: If the class is already defined.
    """
    if name in _RICH_JSON_CLASS_MAPPING:
        logger.warning(f"RichJSON has the class '{name}' already defined")

    _RICH_JSON_CLASS_MAPPING[name] = class_type


def _map_class_by_name(name):
    """
    Internal function to retrieve a mapped class by its name.

    :param name: The name of the mapped class.
    :return: The class reference.
    :raises ValueError: If the class name could not be found.
    """
    if name not in _RICH_JSON_CLASS_MAPPING:
        raise ValueError(
            f"RichJson could not find the class called '{name}'.\nMake sure it is defined in RichJsonClassMapping.")

    return _RICH_JSON_CLASS_MAPPING[name]

File: src/java/main/other/RichJsonClassMapping.java
package other;

import helper.RichJsonLogger;

import java.util.HashMap;
import java.util.Map;

public class RichJsonClassMapping {
    private static final Map<String, Class<?>> RICH_JSON_CLASS_MAPPING = new HashMap<>();

    /**
     * Adds the given class mappings.
     * * @param classMappings A map containing the class names as keys and the class types as values.
     */
    public static void addClassMappings(Map<String, Class<?>> classMappings) {
        for (Map.Entry<String, Class<?>> entry : classMappings.entrySet()) {
            addClassMapping(entry.getKey(), entry.getValue());
        }
    }

    /**
     * Adds the given class to the mapping table.
     * * @param name The name to associate with the class type.
     * @param classType The class type to map.
     * @throws IllegalArgumentException if the mapping name already exists.
     */
    public static void addClassMapping(String name, Class<?> classType) {
        if (RICH_JSON_CLASS_MAPPING.containsKey(name)) {
            RichJsonLogger.logger.warn("has the class '" + name + "' already defined");
        } else {
            RICH_JSON_CLASS_MAPPING.put(name, classType);
        }
    }

    /**
     * Retrieves the mapped class by its name.
     * * @param name The name of the class to retrieve.
     * @return The mapped Class<?> object.
     * @throws IllegalArgumentException if the class name is not found in the mapping.
     */
    public static Class<?> __mapClassByName(String name) {
        if (!RICH_JSON_CLASS_MAPPING.containsKey(name)) {
            throw new IllegalArgumentException("RichJSON could not find the class called '" + name + "'.\nMake sure its defined in RichJsonClassMapping.");
        }
        return RICH_JSON_CLASS_MAPPING.get(name);
    }
}